exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceExporter.java

```diff
@@ -15,134 +15,72 @@
-        configuration.getAttributeMapping(),
-        configuration.getFixedAttributes());
-  }
 
-  private static TraceExporter createWithClient(
-      String projectId,
-      CloudTraceClient cloudTraceClient,
-      ImmutableMap<String, String> attributeMappings,
-      Map<String, AttributeValue> fixedAttributes) {
-    return new TraceExporter(projectId, cloudTraceClient, attributeMappings, fixedAttributes);
-  }
+    public static SpanExporter createWithDefaultConfiguration() {
+        return generateStubTraceExporter(TraceConfiguration.builder());
+    }
 
-  TraceExporter(
-      String projectId,
-      CloudTraceClient cloudTraceClient,
-      ImmutableMap<String, String> attributeMappings,
-      Map<String, AttributeValue> fixedAttributes) {
-    this.projectId = projectId;
-    this.cloudTraceClient = cloudTraceClient;
-    this.projectName = ProjectName.of(projectId);
-    this.translator = new TraceTranslator(attributeMappings, fixedAttributes);
-  }
+    public static SpanExporter createWithConfiguration(TraceConfiguration.Builder configBuilder) {
+        return generateStubTraceExporter(configBuilder);
+    }
 
-  @Override
-  public CompletableResultCode flush() {
-    // We do no exporter buffering of spans, so we're always flushed.
-    return CompletableResultCode.ofSuccess();
-  }
+    @Deprecated
+    public static SpanExporter createWithConfiguration(TraceConfiguration configuration) throws IOException {
+        return InternalTraceExporter.createWithConfiguration(configuration);
+    }
 
-  @Override
-  public CompletableResultCode export(Collection<SpanData> spanDataList) {
-    List<Span> spans = new ArrayList<>(spanDataList.size());
-    for (SpanData spanData : spanDataList) {
-      spans.add(translator.generateSpan(spanData, projectId));
+    @Override
+    public CompletableResultCode flush() {
+        if (internalTraceExporter == null) {
+            return CompletableResultCode.ofFailure();
+        }
+        return internalTraceExporter.flush();
     }
 
-    cloudTraceClient.batchWriteSpans(projectName, spans);
-    return CompletableResultCode.ofSuccess();
-  }
+    @Override
+    public CompletableResultCode export(@Nonnull Collection<SpanData> spanDataList) {
+        try {
+            if (internalTraceExporter == null) {
+                internalTraceExporter = createActualTraceExporter();
+            }
+            return internalTraceExporter.export(spanDataList);
+        } catch (IOException e) {
+            e.printStackTrace();
+            return CompletableResultCode.ofFailure();
+        }
+    }
 
-  @Override
-  public CompletableResultCode shutdown() {
-    this.cloudTraceClient.shutdown();
-    return CompletableResultCode.ofSuccess();
-  }
+    @Override
+    public CompletableResultCode shutdown() {
+        if (internalTraceExporter != null) {
+            return internalTraceExporter.shutdown();
+        } else {
+            return CompletableResultCode.ofFailure();
```

Comment:
Is it actually a failure ot attempt to shutdown something that never started?

Reply:
So this is the case where somehow `shutdown` is called before call to an `export` which initializes the `internalTraceExporter`. 

So it's something that never started. Should we send some other code instead of Failure ?